monasca/templates/api-deployment.yaml

```diff
@@ -51,11 +51,17 @@ spec:
             - name: MYSQL_HOST
               value: "{{ .Release.Name }}-mysql"
             - name: MYSQL_USER
-              value: {{ .Values.api.mysql.user | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-mysql-api-secret"
+                  key: username
             - name: MYSQL_PASSWORD
-              value: {{ .Values.api.mysql.password | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-mysql-api-secret"
+                  key: password
             - name: MYSQL_DB
-              value: {{ .Values.mysql.mysqlDatabase | quote }}
+              value: "mon"
```

Comment:
I've actually removed that property entirely in this patch, since mysql-init doesn't support it. With the charts as they stand right now, changing `mysql.mysqlDatabase` will always lead to a broken configuration.

Given we're deploying our own mysql anyway I figure it's reasonable to be more prescriptive about the database names?